src/main/java/seedu/address/model/visitor/VisitorList.java

```diff
@@ -0,0 +1,51 @@
+package seedu.address.model.visitor;
+
+import java.util.ArrayList;
+import java.util.Objects;
+
+/**
+ * The visitor list to store all the visitors for patient.
+ */
+public class VisitorList extends ArrayList<Visitor> {
+    private ArrayList<Visitor> visitorArrayList;
```

Comment:
Violation of Interface Segregation Principle.
Consider using List<> instead of ArrayList<>.